R/commas_linter.R

```diff
@@ -1,6 +1,7 @@
 #' Commas linter
 #'
-#' Check that all commas are followed by spaces, but do not have spaces before them.
+#' Check that all commas are followed by spaces (e.g. `c(a = 4, b = 5)`),
```

Comment:
I would leave this as a simple sentence, and refer to the style guide link instead. The style guide makes it clearer that the latter example is bad. We could also add more exposition here saying good and bad examples, but that seems redundant to the style guide. WDYT?

Reply:
I am going to revert this for now, but, TBH, I think each linter should have an examples section with at least one example of code that does produce lint and another that doesn't.

I think the style guide is good to provide the underlying rationale for the lint (provided that the reader actually clicks on that link!), but it doesn't lead to much exploration on the reader's part. Giving concrete examples means that the reader can just copy-paste examples and start toying around with them to see what does and what doesn't lint.

If you agree, I can create a new documentation issue to add examples, and can also probably work on it.

Reply:
That's definitely worthy of its own issue. My main concern is about duplicate documentation (if the style guide changes, we need to remember update the docs), but not all of our linters are tidyverse guide enforcements, and those would definitely benefit from a good/bad section.

Reply:
TBH, I don't see the style guide changing that often, so the examples being out of phase with the guide should happen rarely.

I will create a new issue to track this.

Reply:
Done: https://github.com/r-lib/lintr/issues/1492